curator-x-discovery-server/src/main/java/org/apache/curator/x/discovery/server/rest/DiscoveryResource.java

```diff
@@ -90,7 +90,7 @@ public Response     putService(ServiceInstance<T> instance, @PathParam("name") S
             return Response.status(Response.Status.BAD_REQUEST).build();
         }
         
-        if ( instance.getServiceType() == ServiceType.DYNAMIC )
+        if ( instance.getServiceType() == ServiceType.DYNAMIC || instance.getServiceType() == ServiceType.DYNAMIC_SEQUENTIAL )
```

Comment:
Maybe we can add an `.isDynamic()` method to `ServiceType` and call that instead of checking against these two specific cases?